docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,209 @@
+
+<!-- image source diagrams: https://docs.google.com/presentation/d/1lho38WMS3spo0euY_l_jsrknSeOqJkeGheMohI7R7F0/edit#slide=id.g1ffd17cda3_0_123 -->
+
+*   Kubectl plugins extend the kubectl binary ①.  [Link for 1.8 docs branch: docs/tasks/extend-kubectl/kubectl-plugins ] They only affect the individual user's local environment.  These extension points are described in the [Client Extensions](todo) section.
+*   The apiserver ②  handles all requests.  Several types of extension points in the apiserver allow authenticating requests, or blocking them based on their content, editing content, and handling deletion.  These are described in the [API Access Extensions](todo) section. 
+*   The apiserver serves various kinds of **resources**, ③.  **Built-in resource kinds**, like `pods`, are defined by the Kubernetes project and have to be .     You can also add resources that you definde are several ways to add **Custom Resources** to your API-Server.  Custom Resources are often used with API Access Extensions.
+*   The kubernetes scheduler, ④, decides which nodes to place pods on.  There are several ways to extend scheduling.  These are described in the [Scheduler Extensions](bloop) section.
+*   Much of the behavior of Kubernetes is implemented by client programs Controllers, ⑤.  You can write your own, as described in the [Controllers](boo) section.  Controllers are often used in conjunction with Custom Resources.
+*   The kubelet runs on servers, and mediates access to the network, ⑥, and storage, ⑦.  These are described in the [Infrastructure Extensions](poof) section.
+
+If you are unsure where to start, this flowchart can help.  Note that some solutions may involve several types of extensions.
+
+
+
+<img source="https://docs.google.com/drawings/d/e/2PACX-1vRWXNNIVWFDqzDY0CsKZJY3AR8sDeFDXItdc5awYxVH8s0OLherMlEPVUpxPIB1CSUu7GPk7B2fEnzM/pub?w=1440&h=1080">
+
+<!-- image source drawing: https://docs.google.com/drawings/d/1sdviU6lDz4BpnzJNHfNpQrqI9F19QZ07KnhnxVrp2yg/edit --> 
+
+## API Extensions
+### User-Defined Types
+
+Consider adding a Custom Resource to Kubernetes if you want to define new controllers, application configuration objects  or other declarative APIs, and to manage them using Kubernetes tools, such as `kubectl`.  
+
+Do not use a Custom Resource as data storage for application, user, or monitoring data.
+
+For more about Custom Resources, see the [Custom Resources concept guide](/docs/concepts/api-extension/custom-resources.md).
+
+
+### Combining New APIs with Automation
+
+Often, when you add a new API, you also add a control loop that reads and/or writes the new APIs.  This is often called the **Operator** pattern.
+
+### Changing Built-in Resources
+
+When you extend the Kubernetes API by adding custom resources, the added Kinds always fall into new API Groups.  You cannot replace or change existing API groups.
```

Comment:
Suggest either:
...the added **Kind** always falls into a new API Group.
with kind definition.  Or same edit but "kind" instead of "**Kind**".

Reply:
You can add several kinds to a new api group.  I've used the word resource instead.

Reply:
✔️